### DISCUSSION TITLE
Does anyone have a problem with staleTime: Infinity in react native app?

### DISCUSSION THREAD

**Discussion by stevan-borus** [TanStack/query#7568] [Q&A] [answered]
When I set `staleTime` to Infinity for a query in a react native app it just falls back to null for some reason. 
If I set `staleTime` to some number it picks it up and works correctly.

This doesn't happen in a web app for that same query.

<img width="462" alt="Screenshot 2024-06-14 at 18 10 30" src="https://github.com/TanStack/query/assets/17165251/354ee7c4-a158-4913-bbc7-953345602f6f">
<img width="328" alt="Screenshot 2024-06-14 at 18 08 37" src="https://github.com/TanStack/query/assets/17165251/340fd47b-aae2-41a1-8175-1687fe7dc185">
<img width="462" alt="Screenshot 2024-06-14 at 18 10 19" src="https://github.com/TanStack/query/assets/17165251/915f3f3c-9c85-40af-b4cf-b654b7955d86">
<img width="328" alt="Screenshot 2024-06-14 at 18 08 45" src="https://github.com/TanStack/query/assets/17165251/398b6211-df53-48c0-be83-49c8d568f434">

**Comment by TkDodo** [comment, marked as answer]
where are those screenshots coming from? Could it be that whatever shows you this just cannot serialize `Infinity` correctly and just shows null? I mean it's what you get when you do `JSON.stringify`:

![Screenshot 2024-06-21 at 10 44 53](https://github.com/TanStack/query/assets/1021430/96b6c558-df05-490e-9b06-e82a7de2922f)

So I don't think it really is null - it's just shown as null.

**Reply by stevan-borus** [reply]
Yup, that is the problem.
The screenshots are from react query dev tools window from https://github.com/expo/dev-plugins
